tests/Unit/OauthRegistrationSettingsTest.php

```
@@ -0,0 +1,57 @@
+ *
+ * These tests verify that the OAuth callback correctly handles
+ * the is_oauth_registration_enabled setting in combination with
+ * the is_registration_enabled setting.
+ */
+
+describe('OAuth Registration Settings Logic', function () {
+    /**
+     * Helper to check if registration should be allowed based on settings.
+     * This mirrors the logic in OauthController::callback()
+     */
+    function shouldAllowOAuthRegistration(bool $generalRegistrationEnabled, bool $oauthRegistrationEnabled): bool
```

PHPMD (2.15.0)

[warning] Avoid excessively long variable names like $generalRegistrationEnabled. Keep variable name length under 20. (undefined)

(LongVariable)